parser/parser.go

```diff
@@ -873,12 +874,18 @@ func (p *parser) expandMacro(exprID int64, function string, target *exprpb.Expr,
 	eh.parserHelper = p.helper
 	eh.id = exprID
 	expr, err := macro.Expander()(eh, target, args)
+	// An error indicates that the macro was matched, but the arguments were not well-formed.
 	if err != nil {
 		if err.Location != nil {
 			return p.reportError(err.Location, err.Message), true
 		}
 		return p.reportError(p.helper.getLocation(exprID), err.Message), true
 	}
+	// An empty Expr value from the macro indicates that the macro implementation decided that
+	// an expansion should not be performed.
```

Comment:
Wouldn't returning nil be more idiomatic for not-present pointer value?

Reply:
I naively copied the behavior from cel-cpp without thinking about what would be idiomatic. Updated.